### PR TITLE
update to go 1.21

### PR DESCRIPTION
**Pull request by faddat**
## 1. Summary
This PR updates golang because we should do that every time we update the chain

## 2.Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## 3. Implementation details

Just bump golang

### CODE REVIEW

**Comment by faddat** [comment]
@dependabot recreate

File: Makefile [hunk 1]
```diff
@@ -130,7 +130,7 @@ endif
 BUILD_TARGETS := build install
 
 check_version:
-ifneq ($(GO_MINOR_VERSION),20)
+ifneq ($(GO_MINOR_VERSION),21)
 	@echo "ERROR: Go version 1.20 is required for building Quicksilver. There are consensus breaking changes between binaries compiled with different Go versions."
```
Comment (joe-bowman) [inline Makefile:134]:
```suggestion
	@echo "ERROR: Go version 1.21 is required for building Quicksilver. There are consensus breaking changes between binaries compiled with different Go versions."
```

**Review by joe-bowman — DISMISSED** [review]
lgtm

**Comment by faddat** [comment]
merging since we keep invalidating reviews by merging in main.  Will adjust github settings to be less spicy.

**Pull request merged**